### PR TITLE
DB Anonymization

### LINKED ISSUE #3878: Create masking rules and script to anonymize data from any simple report database
Using: https://gitlab.com/dalibo/postgresql_anonymizer

Document and create masking rules for all fields that may contain sensitive information.

This requires valid users and org ids for the app to function correctly due to our integration with Okta.

The faker database will be our source of truth for fake and real data(only emails and org ids). Populate that database with any additional data we need. 

Pg restore command should restore the anonymized dump to our target database

### PR DESCRIPTION
**Pull request by alismx**
# DATABASE PULL REQUEST

## Related Issue

- Resolves #3878 
- We want to anonymize databases for testing against realistic databases.

## Changes Proposed

- Add a way to generate data to load into our anonymized db (not used yet)
- Add a way to anonymize a PostgreSQL db dump
- Add masking rules for all potentially sensitive columns and json blobs in our db

## Additional Information

- The data generation script is a WIP and is not in use, although it has been practically tested.
- Would love to figure out how to test our masking rules in an automated way and enforce having a masking rule for each column
- Should write tests for our data generation script but that was left out as it may be stripped or drastically changed.
- #3879 and #3289 are upcoming and will expand on this. Extensive testing will take place in lower environments.

## Testing

- Run the yarn commands while your local docker containers are running
- **Check the masking rules.**

## Checklist for Primary Reviewer

- [ ] Only database changes are included in this PR
- [ ] Any new tables or columns that do not contain PII are accompanied by a GRANT SELECT to the no-PHI user
- [ ] Any changes to tables that have custom no-PHI views are accompanied by changes to those views (including re-granting permission to the no-PHI user if need be)
- [ ] Each new changeset has a corresponding [tag](https://docs.liquibase.com/change-types/community/tag-database.html)
- [ ] Rollback has been verifed locally and in a deployed environment
- [ ] Masking rules have been verified
----

### CODE REVIEW

File: backend/db-setup/README.md
Comment (DanielSass) [inline backend/db-setup/README.md:53]: typo?
Our script ~it~ is working

File: backend/db-setup/README.md
Comment (nathancrtr) [inline backend/db-setup/README.md:53]: Neat, looks like they just merged a fix too! 🎉

File: docker-compose.yml
Comment (nathancrtr) [inline docker-compose.yml:24]: Is this a stray?

Comment (alismx) [reply docker-compose.yml:24]: Hmm, I don't remember commenting this line. I'll put this back and thoroughly test before pushing.

Comment (alismx) [reply docker-compose.yml:24]: I don't use the `SR_DB_PORT` variable, but removing this would cause issues for anybody using it. I'm confident that `5432` is the default anyway so that behavior wouldn't change. A stray, I believe it is!

Comment (alismx) [reply docker-compose.yml]: Uncommented this line and pushed!

Comment (nathancrtr) [reply docker-compose.yml]: > I don't use the SR_DB_PORT variable, but removing this would cause issues for anybody using it

This would explain why my local DB setup blew up on this branch 🙂 thanks!

File: backend/db-setup/generate-db-data.sh
Comment (mehansen) [inline backend/db-setup/generate-db-data.sh:7]: typo: out = our?

File: backend/db-setup/generate-db-data.sh
Comment (rin-skylight) [inline backend/db-setup/generate-db-data.sh:1]: Is it worth being able to take in a variable value that determines how many lines to generate? Say, if we wanted to simulate a local copy of prodlike data?

File: backend/db-setup/generate_db_data.py
Comment (mehansen) [inline backend/db-setup/generate_db_data.py:34]: some of these values don't have generator methods defined
e.g. `generate_db_data.py --table street_name` is going to fail

File: backend/db-setup/anonymize-db.sh
Comment (rin-skylight) [inline backend/db-setup/anonymize-db.sh:6]: Any chance you can add in an additional `echo` statement here that lets the user know you're prompting for `sudo` access? I spent a little too much time trying to dig up database passwords before I realized this was a `sudo` request. 😅

Comment (alismx) [reply backend/db-setup/anonymize-db.sh]: done

File: backend/db-setup/generate-db-data.sh
Comment (alismx) [reply backend/db-setup/generate-db-data.sh]: Doing this would be fine, but this file no longer exists and has been replaced by a sample SQL file that shows how to fill the faker tables. Let's revisit this request when it becomes more than an example.

File: backend/db-setup/generate_db_data.py
Comment (alismx) [reply backend/db-setup/generate_db_data.py]: Thanks for pointing this out, should be all resolved now.

**Review by DanielSass — APPROVED** [review]

**Comment by alismx** [comment]
Any chance I can get a second review on here? I'd love to get this closed out so it can be iterated on, and I can move forward to using it in a remote environment!

After it merges, I'd like to break out the small readme into a wiki page.

**Review by rin-skylight — APPROVED** [review]
Thank you for the amazing work on this! 🚀 🎉

**Pull request merged**
